File: backend/src/main/java/fr/pedalons/dto/places/response/PlaceListResponse.java
package fr.pedalons.dto.places.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated place list response")
@ValidateSchema
public record PlaceListResponse(
    @Schema(description = "List of places", required = true) List<PlaceDetailDto> places,
    @Schema(description = "Total number of places", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/placeListResponse.ts
import type { PlaceDetailDto } from './placeDetailDto.ts'

/**
 * Paginated place list response
 */
export interface PlaceListResponse {
  /** List of places */
  places: PlaceDetailDto[]
  /** Total number of places */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
